File: Number_Conversion/Java/Decimal_to_binary.java
//JAVA program to convert Decimal integer to it's Binary form

import java.util.Scanner;

public class Decimal_to_binary {

    //Driver Code
    public static void main(String[] args){
        int n, rem;
        String Bin = "";    //String to store the binary number
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter any decimal number:");
        n = sc.nextInt();   //Taking the decimal input

        if (n == 0){
            Bin = "0";
        }

        //Simultaneously storing the remainder when number divided by 2 in the string in reverse order
        while (n > 0){
            rem = n % 2;
            Bin = rem + Bin;
            n = n / 2;
        }
        System.out.println("Binary number:"+Bin);   //output
    }
}
//Input : 10 
//Output : 1010 

File: Number_Conversion/C++/Decimal_to_Binary.cpp

// C++ program to convert numbers from decimal to binary form

#include <iostream>
#include <math.h>
using namespace std;

// Function to convert the number from decimal to binary form
void printBinary(int n){

    // command to find out number of bits in the given number
    // which decides the size of the array
    int size = log2(n) + 1;
    
    // Array to store the binary number    
    int b[size], j = 0;    
    while(n != 0){
        b[j] = n % 2;
        n = n / 2;
        j++;
    }

    cout << "The binary form is : ";
    for(j = size - 1; j >= 0; j--)
        cout << b[j];
}


int main()
{
    cout << "Enter the number you wish to convert:" << endl;
    int n;
    cin >> n;
    printBinary(n); // Calling function printBinary to convert the number to binary form
    cout << "\n";
    return 0;
}
/*
Input : 25
Output : 11001
*/
